### DISCUSSION TITLE
Controls in the subcanvas cannot be displayed

### DISCUSSION THREAD

**Discussion by Maklith** [AvaloniaUI/Avalonia#14331] [Q&A] [answered]
I have some controls and his visual tree looks like this
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/fe55bfb6-2d85-4675-af28-6efb6700c5d8)
But the control circled in the figure doesn't display properly, even though I can see it in devtools
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/37ea0a82-24fe-453e-b1f2-6a32d8f86ee9)
Strangely enough, if I use a grid instead of a canvas, it will display well even though it can't display in negative coordinates
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/5833ded2-8672-453e-9f81-a5a32dc634e4)
It's worth mentioning that I had to use Canvas because Border needs to follow the mouse at all times

Project source code address:
[Nodify.Avalonia](https://github.com/MakesYT/Nodify.Avalonia)
So my question is to know if I'm doing things wrong ? 

Thanks in advance.

**Comment by stevemonaco** [comment]
The `Canvas` in `PendingConnection.axaml` needs some dimensions. I hackfixed this with the following:
```xml
<Canvas
    Width="{Binding $parent[Canvas].Bounds.Width}"
    Height="{Binding $parent[Canvas].Bounds.Height}"
    x:DataType="Canvas">
```

I do have to question the overall design of so many Canvas controls being stacked or nested, but maybe it will work out.

Besides that, there's another small bug in `PendingConnections.axaml.cs` where you cast to `Brush` instead of `IBrush`. I was getting a DevTools crash when an `ImmutableSolidColorBrush` was involved.

```xml
public IBrush Stroke
{
    get => (Brush)GetValue(StrokeProperty);
    set => SetValue(StrokeProperty, value);
}
```

**Reply by Maklith** [reply]
Unfortunately, it still can't work in negative coordinates
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/6ebb5061-51a4-4770-ba1c-b675d8f2b8ce)

**Reply by Maklith** [reply, marked as answer]
I solved it.
```xaml
<Canvas  Width="{Binding $parent[Canvas].Bounds.Width}" RenderTransform="{Binding $parent[controls:NodifyEditor].ViewTranslateTransform}"
                         Height="{Binding $parent[Canvas].Bounds.Height}"
                         x:DataType="Canvas">
```
